include/bout/mesh.hxx

```diff
@@ -636,6 +636,19 @@ public:
     return inserted.first->second;
   }
 
+  std::shared_ptr<Coordinates>
+  getCoordinatesConst(const CELL_LOC location = CELL_CENTRE) const {
```

Comment:
I think ideally this would be:

```suggestion
  const Coordinates&
  getCoordinates(const CELL_LOC location = CELL_CENTRE) const {
```

but maybe this is too likely to cause issues elsewhere